src/components/Deck.js

```diff
@@ -1,11 +1,20 @@
 import React from 'react'
 
 import Card from './Card'
+import DummyCard from './DummyCard'
 import './Deck.css'
 
-function Deck({ onSwipe, cards }) {
+function Deck({ onSwipe, cards, tick = 0 }) {
     return (
         <div className="deck-root">
+            {[0, 1, 2, 3, 4, 5].map((key, index, list) => (
```

Comment:
Nvm, re-renders seems to be caused by the `tick` (which is controlled by `game.state.rounds`). `React.useMemo()` prevents the rotation changes and significantly improves perceived performance on my phone, but will ignore `tick` unless you add it as a dependency for when to update the memo.

The core problem is really that `DummyCard`s change rotation on every `tick`.

I really like the effect of cards being added to the back of the stack, but it would be even better if `DummyCard`s keep their rotation as long as they are present in the view. Instead, perhaps we could add a new `DummyCard` to the bottom and remove the top one (just under the real card), to make the card stack look like it's moving up?

Reply:
Could you explain what you use useMemo for? Do you mean that the the card pile remains static? In that case I don't really see the use case for useMemo. The computations of these six cards should by no means be heavy.

I do understand that the perceived performance could improve when ignorning tick though since it would reduce the laggy feeling. I think the problem is related to react not wanting to update components when they are rendered in another order even though they retain their keys. Which is why the only animated card is the first and last card.

This could be solved by using z-indices but I'd rather not. So if you have any suggestions I am happy to listen.

Reply:
And thanks for the input.

Reply:
I tried [`useMemo()`](https://reactjs.org/docs/hooks-reference.html#usememo) to memoize the rendered list of components and effectively preventing re-renders.

## We'll have to discuss the following as I'm not entirely sure, but here's an idea:

One alternative to improve performance would be to go with the swiping implementation used in Lapse (https://www.youtube.com/watch?v=fxPLD4wmmJw)

It's much simpler (both to implement and for performance):
- Show card background exactly behind the current swipable card.
- Dragging a card and releasing it will move it back to it's original position, completely covering the background again.
- Only keep one card in the deck that is swipable at any given time.
- Show the next card without a flipping animation.

## Advantages
1) A simpler single card stack shows players that the next card doesn't exist long before - it's selected after each swipe. This fits with the game setting (you don't know what's happening tomorrow)
2) Keeping animations really simple puts focus on the game content itself rather than the card stack.
3) Easier to implement
4) Easier to get good performance

## Disadvantages
1) The opposite of advantage 1) could be said too: A more advanced, realistic card stack would show the "pile of papers" on the presidents desk. You don't know exactly what's on your next task until you turn the page, but you can see you've got things to do.
2) This simple "single card stack" won't be as nice looking as the the current card stack of this PR. Even though it takes some tweaking to get right, this might be worth the effort to give the game a more unique look and feel.